docs/DeveloperGuide.md

```diff
@@ -926,26 +927,65 @@ For all use cases below, the **System** is `bookmark` and the **Actor** is the `
 1. User requests to delete a note from a specific book using its displayed index.
 2. The requested note is deleted from the specified book.
 
+    Use case ends.
+
 **Extensions**
 
 * 1a. The index given is invalid.
 
     * 1a1. bookmark returns an error message.
 
-    Use case resumes at step 1.
+    Use case ends.
     
 * 1b. The note index provided is invalid.
 
     * 1b1. bookmark returns an error message.
     
-    Use case resumes at step 1.
+    Use case ends.
 
 **Use Case: UC11 - Sorting the book list**
 
 **MSS**
 
-1. User requests to sort the book list by name.
-2. bookmark sorts all the books by name.
+1. User requests to sort the displayed book list by a specified field.
+2. bookmark sorts the displayed book list by the specified field.  
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. The field provided is invalid.
+
+    * 1a1. bookmark returns an error message.
+
+    Use case ends.
+    
+**Use Case: UC12 - Finding keywords**
+
+**MSS**
+
+1. User requests to find keyword(s) in the displayed book list by a specified field.
+2. Books that contain keyword(s) in the specified field are returned to the user in a new list.
+
+    Use case ends.
+    
+**Extensions**
+
+* 1a. The field provided is invalid.
+
+    * 1a1. bookmark returns an error message.
+
+    Use case ends.
+
+ * 1b. bookmark does not find any matching books, but notes that there are similar words to the input keyword(s) in the displayed book list. 
+
+    * 1b1. bookmark returns a message informing user that no matching books were found and suggests an alternative keyword to the user that is similar to the keyword(s) inputted.
+
+    Use case ends.
+
+ * 1c. bookmark does not find any matching books, and notes that there no similar words to the input keyword(s) in the displayed book list. 
+
```

Comment:
matching books, but notes. (Comma not needed.)

any matching books, and notes (Comma not needed)

Will it be better to rephrase the notes that there are similar words, eg. talk about the suggestion more explicitly instead of notes